test/integration/middleware_test.rb

```diff
@@ -0,0 +1,22 @@
+require 'integration_test_helper'
+
+class MiddlewareIntegrationTest < ActionDispatch::IntegrationTest
+  test "secure headers are set" do
+    visit '/'
+    assert_equal page.response_headers['X-Frame-Options'], 'sameorigin'
+    assert_equal page.response_headers['X-XSS-Protection'], '1; mode=block'
+    assert_equal page.response_headers['X-Content-Type-Options'], 'nosniff'
+    assert_equal page.response_headers['Content-Security-Policy'], \
+      "default-src 'self'; child-src 'self'; connect-src 'self' ws: wss:; "+
+      "img-src 'self' *.gravatar.com; script-src 'unsafe-eval' 'unsafe-inline' "+
+      "'self'; style-src 'unsafe-inline' 'self'"
+  end
+
+  test "webpack dev server adds the dev server to Content-Security-Policy" do
+    Rails.configuration.webpack.dev_server.enabled = true
+    Webpack::Rails::Manifest.stubs(:asset_paths).returns([])
+    visit '/'
+    assert page.response_headers['Content-Security-Policy'].include?(Rails.configuration.webpack.dev_server.port.to_s)
+    Rails.configuration.webpack.dev_server.enabled = false
```

Comment:
Nitpick, please put this line into an `ensure` block - if the test ever fails, it prevents other tests from being affected, making the root cause a lot clearer.